var/spack/repos/builtin/packages/py-protobuf/package.py

```diff
@@ -91,6 +91,8 @@ class PyProtobuf(PythonPackage):
         else:
             conflicts("+cpp", when="@2." + str(ver))
 
+
+class PythonPipBuilder(spack.build_systems.python.PythonPipBuilder):
     @property
```

Comment:
Do build settings for packages _have_ to go into a Builder class? I'm not opposed to this, but it's a big change that affects thousands of packages. I'm guessing it's optional, but I'm wondering if we want to make that the suggested/recommended behavior or if we want to reserve that only for multi-build system packages only.

Reply:
> Do build settings for packages have to go into a Builder class?

Yes, that's the change this PR does - extract everything related to the installation procedure and move it into a separate class. Note that the division is not "final" and if we see that further methods should be moved into builder we might do it in a following PR.

Most of the current packages will still work, for the time being, since we generate at run-time a builder that forwards method calls to the old-style package.